### PR TITLE
toolbox: init at 0.0.99.3

### LINKED ISSUE #96115: Toolbox
**Project description**

It's a tool to create pet-container using Podman. The container is fully mutable and maps your user home to the container. 

**Metadata**

* homepage URL: https://github.com/containers/toolbox
* source URL: https://github.com/containers/toolbox
* license: Apache-2.0
* platforms: Linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

The two fetched patches from #110473 have been merged upstream, so they have been dropped, but we have carried through the glibc patch. It sounds like the ux of this is still pretty sharp, but I am happy to try and drive this one forward.

Fixes #96115
Closes #110473

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/toolbox/glibc.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
++	glibcMountArg := "@glibc@:@glibc@:ro"
++
+ 	var runtimeDirectory string
+ 	var xdgRuntimeDirEnv []string
+ 
+@@ -423,6 +425,7 @@ func createContainer(container, image, release string, showCommandToEnter bool)
+ 		"--volume", toolboxPathMountArg,
+ 		"--volume", usrMountArg,
+ 		"--volume", runtimeDirectoryMountArg,
++		"--volume", glibcMountArg,
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/toolbox/glibc.patch:18]: We could inline this

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10674,6 +10674,7 @@ with pkgs;
   pocketbase = callPackage ../servers/pocketbase { };
 
   podman = callPackage ../applications/virtualization/podman/wrapper.nix { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10677]:
```suggestion
```

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 203904` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>toolbox</li>
</ul>
</details>

File: pkgs/applications/virtualization/toolbox/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  nativeBuildInputs = [ go-md2man installShellFiles ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/containers/toolbox/pkg/version.currentVersion=${version}"
+  ];
+
+  preCheck = "export PATH=$GOPATH/bin:$PATH";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/toolbox/default.nix:32]:
```suggestion
  preCheck = ''
    export PATH=$GOPATH/bin:$PATH
  '';
```

**Pull request merged**